test/regress/cli/regress0/quantifiers/issue9602-rewrite-pow-type.smt2

```diff
@@ -0,0 +1,4 @@
+; EXPECT: sat
+(set-logic ALL)
+(assert (forall ((t Int)) (<= 0.0 (* 1.0 (^ 0 1.0)))))
```

Comment:
I suggest having two more tests: one in which the base is some "normal number" (say 3 or 5) and one where the base is 2 (as this would be re-written to `pow2`, that might also introduce these typing issues). @